sources/packages/api/src/esdc-integration/disbursement-receipt-integration/disbursement-receipt-files/disbursement-receipt-file-record.ts

```diff
@@ -0,0 +1,47 @@
+import { getDateOnlyFromFormat } from "../../../utilities";
+import { DATE_FORMAT } from "../../models/esdc-integration.model";
+import { DisbursementReceiptRecordType } from "../models/disbursement-receipt-integration.model";
+
+/**
+ * Base class for disbursement receipt record.
+ * All the records of disbursement receipt must be a sub class of this.
+ */
+export class DisbursementReceiptRecord {
+  constructor(
+    protected readonly line: string,
+    protected readonly _lineNumber = 0,
+  ) {}
+
+  public get lineNumber(): number {
+    return this._lineNumber;
+  }
+
+  public get recordType(): DisbursementReceiptRecordType {
+    return this.line.substring(10, 11) as DisbursementReceiptRecordType;
+  }
+
+  /**
+   * Any date record as string is parsed to date object.
+   * @param dateText
+   * @returns Date object.
+   */
+  protected convertToDateRecord(dateText: string) {
+    return getDateOnlyFromFormat(dateText, DATE_FORMAT);
+  }
+
+  /**
+   * Parses any amount field to numeric string by ignoring any junk character is present in data.
```

Comment:
I got the idea of a workaround for the current garbage characters received in the file. Just keep in mind that we follow the ticket orientation and read only the main part (0, 5) this method will be not necessary at all.